examples/ecs/fetch.rs

```diff
@@ -0,0 +1,93 @@
+use bevy::ecs::component::Component;
```

Comment:
At the start of this example, I'd like to discuss the use cases and motivation for this feature a bit.

I would also demonstrate the ability to use type aliases in many trivial cases, and explain when you need to go beyond that.

Reply:
I've updated the example. I'm starting to doubt whether it's not too much though, speaking of its complexity.